apps/ideaspace/src/components/common/IdeaForm/IdeaForm.js

```diff
@@ -7,227 +7,236 @@ import SubmissionButton from './SubmissionButton';
 import EditionButton from './EditionButton';
 import Dropdown from '@devlaunchers/components/components/organisms/Dropdown';
 import useResponsive from '@devlaunchers/components/src/hooks/useResponsive';
-import Checkbox from '@devlaunchers/components/src/components/Checkbox/Checkbox'
+import Checkbox from '@devlaunchers/components/src/components/Checkbox/Checkbox';
 
-const IdeaForm = ({
-	initialValues,
-	SignupSchema,
-	submitHandler,
-	unsavedHandler,
-	formButton,
-	sending,
-	clickHandler
-},props) => {
-	const {	errors } = props;
-	const [disabling, setDisabling] = React.useState(true);
-	const { isMobile } = useResponsive();
-	
-	const compareValuesToInitial = (values) => {
-		const name = Object.keys(values);
-		for (let i = 0; i < name.length; i++) {
-			if (values[name[i]] !== initialValues[name[i]]) {
-				return true;
-			}
-		}
-		return false;
-	}
+const IdeaForm = (
+  {
+    initialValues,
+    SignupSchema,
+    submitHandler,
+    unsavedHandler,
+    formButton,
+    sending,
+    clickHandler,
+  },
+  props
+) => {
+  const { errors } = props;
+  const [disabling, setDisabling] = React.useState(true);
+  const { isMobile } = useResponsive();
 
-	const AutoSubmitToken = () => {
-		const { values, submitForm } = useFormikContext();
-		React.useEffect(() => {
-			if (compareValuesToInitial(values)) {
-				unsavedHandler(true);
-				setDisabling(false);
-			} else {
-				unsavedHandler(false);
-				setDisabling(true);
-			}
-		}, [values]);
-		return null;
-	};
-	return (
-		<atoms.Box margin='1rem 1.5rem 3rem 1.5rem'>
-			<atoms.Box maxWidth='36rem' margin='auto' style={{ textAlign: 'left' }}>
-			<Formik
-				initialValues={initialValues}
-				validationSchema={SignupSchema}
-				onSubmit={submitHandler}
-				enableReinitialize
-			>
-				{({ errors, setFieldValue, touched }) => (
-					<Form>
-						<AutoSubmitToken />
-						<atoms.Typography type='h4'>
-							Idea Info
-							<hr noshade="false" style={{margin: "1rem auto 2rem"}}/>
-						</atoms.Typography>
+  const compareValuesToInitial = (values) => {
```

Comment:
Can you give an example of what are we comparing?

Reply:
@chungthuang this is to check if any form values have been updated.